.github/workflows/sonarqube.yml

```
@@ -0,0 +1,42 @@
+          python-version: '3.8'  # Change this to the version you are using
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pytest pytest-cov
+
+      - name: Run tests with coverage
+        run: |
+          pytest --cov=backend --cov-report=xml  # Ensure the coverage report is generated
+          
```

YAMLlint (1.35.1)

[error] trailing spaces

(trailing-spaces)